### ISSUE TITLE
HW: rabbit_framing.hrl redefined in amimulator_amqp.erl

### ISSUE THREAD

**Issue by busseyl** [voxter/kazoo#16]
[root@lucas01 amimulator]# make clean
rm -f ebin/*
rm -f test/_.beam test/amimulator.app
rm -f erl_crash.dump
[root@lucas01 amimulator]# make
erlc -v -Werror +debug_info +warn_export_all -I../../core -I../../deps -pa ../../core/kazoo_ips-1.0.0/ebin -pa ../../core/webseq-1.0.0/ebin -pa ../../core/kazoo_oauth-1.0.0/ebin -pa ../../core/kazoo_caches-1.0.0/ebin -pa ../../core/whistle_config-1.0.0/ebin -pa ../../core/kazoo_sip-1.0.0/ebin -pa ../../core/whistle_couch-1.0.0/ebin -pa ../../core/whistle_stats-1.0.0/ebin -pa ../../core/kazoo_token_buckets-1.0.0/ebin -pa ../../core/braintree-1.0.0/ebin -pa ../../core/whistle_apps-1.0.0/ebin -pa ../../core/whistle_services-1.0.0/ebin -pa ../../core/kazoo_etsmgr-1.0.0/ebin -pa ../../core/whistle_amqp-1.0.0/ebin -pa ../../core/kazoo_modb-1.0.0/ebin -pa ../../core/kazoo_documents-1.0.0/ebin -pa ../../core/kazoo_translator-1.0.0/ebin -pa ../../core/whistle-1.0.0/ebin -pa ../../core/whistle_number_manager-1.0.0/ebin -pa ../../core/whistle_media-1.0.0/ebin -pa ../../core/kazoo_xml-1.0.0/ebin -pa ../../core/whistle_transactions-1.0.0/ebin -pa ../../core/kazoo_bindings-1.0.0/ebin -pa ../../deps/merl-28e5b3829168199e8475fa91b997e0c03b90d280/ebin -pa ../../deps/gproc-0.2.12/ebin -pa ../../deps/wsock-1.1.5/ebin -pa ../../deps/erlsom-1.3/ebin -pa ../../deps/lager_syslog-1.2.2/ebin -pa ../../deps/socketio-0.1.0/ebin -pa ../../deps/detergent-0.2/ebin -pa ../../deps/eiconv-644fb5e7bd6640fbd073f4d28957914ea979aea0/ebin -pa ../../deps/folsom-0.8.0/ebin -pa ../../deps/lager-2.0.0/ebin -pa ../../deps/erlydtl-0.9.4/ebin -pa ../../deps/rabbit_common-3.3.5/ebin -pa ../../deps/ecsv-1/ebin -pa ../../deps/rabbitmq_client-3.3.5/ebin -pa ../../deps/recon-1.2.0/ebin -pa ../../deps/cowboy-0.10.0/ebin -pa ../../deps/zucchini-1.0.0/ebin -pa ../../deps/slex-0.2.2/ebin -pa ../../deps/couchbeam-0.8.0/ebin -pa ../../deps/mochiweb-2.4.2/ebin -pa ../../deps/jesse-1.1.5/ebin -pa ../../deps/erlang_localtime-1.0.1/ebin -pa ../../deps/gen_smtp-8f7ba01172e679e8b4262334cefc3451b4c36777/ebin -pa ../../deps/cowlib-0.6.2/ebin -pa ../../deps/ibrowse-4.0.1/ebin -pa ../../deps/ranch-0.10.0/ebin -pa ../../deps/rabbitmq_server-3.3.5/ebin -pa ../../deps/exmpp-0.9.9/ebin -pa ../../deps/nksip-0.4/ebin -pa ../../deps/bear-0.8.0/ebin -pa ../../deps/meck-0.7.2/ebin -pa ../../deps/oauth-1.2.2/ebin -pa ../../deps/goldrush-0.1.2/ebin -pa ../../deps/gen_leader-0.0.1/ebin -pa ../../deps/ejson-0.1.0/ebin -pa ../../deps/erlang-syslog-1.0.1/ebin -pa ../../deps/poolboy-1.1.0/ebin -pa ../../deps/edown-0.0.1/ebin -o ebin/ -pa ebin/ src/amimulator_amqp.erl src/amimulator_app.erl src/amimulator_commander.erl src/amimulator_comm.erl src/amimulator.erl src/amimulator_serv.erl src/amimulator_sup.erl src/amimulator_util.erl src/modules/amimulator_acdc.erl src/modules/amimulator_call.erl src/modules/amimulator_originator.erl
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:18: redefining macro 'PROTOCOL_PORT'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:19: redefining macro 'FRAME_METHOD'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:20: redefining macro 'FRAME_HEADER'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:21: redefining macro 'FRAME_BODY'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:22: redefining macro 'FRAME_HEARTBEAT'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:23: redefining macro 'FRAME_MIN_SIZE'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:24: redefining macro 'FRAME_END'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:25: redefining macro 'REPLY_SUCCESS'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:26: redefining macro 'CONTENT_TOO_LARGE'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:27: redefining macro 'NO_ROUTE'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:28: redefining macro 'NO_CONSUMERS'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:29: redefining macro 'ACCESS_REFUSED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:30: redefining macro 'NOT_FOUND'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:31: redefining macro 'RESOURCE_LOCKED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:32: redefining macro 'PRECONDITION_FAILED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:33: redefining macro 'CONNECTION_FORCED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:34: redefining macro 'INVALID_PATH'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:35: redefining macro 'FRAME_ERROR'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:36: redefining macro 'SYNTAX_ERROR'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:37: redefining macro 'COMMAND_INVALID'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:38: redefining macro 'CHANNEL_ERROR'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:39: redefining macro 'UNEXPECTED_FRAME'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:40: redefining macro 'RESOURCE_ERROR'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:41: redefining macro 'NOT_ALLOWED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:42: redefining macro 'NOT_IMPLEMENTED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:43: redefining macro 'INTERNAL_ERROR'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:44: redefining macro 'FRAME_OOB_METHOD'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:45: redefining macro 'FRAME_OOB_HEADER'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:46: redefining macro 'FRAME_OOB_BODY'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:47: redefining macro 'FRAME_TRACE'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:48: redefining macro 'NOT_DELIVERED'
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:50: record 'connection.start' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:51: record 'connection.start_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:52: record 'connection.secure' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:53: record 'connection.secure_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:54: record 'connection.tune' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:55: record 'connection.tune_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:56: record 'connection.open' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:57: record 'connection.open_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:58: record 'connection.close' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:59: record 'connection.close_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:60: record 'connection.blocked' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:61: record 'connection.unblocked' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:62: record 'connection.redirect' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:63: record 'channel.open' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:64: record 'channel.open_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:65: record 'channel.flow' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:66: record 'channel.flow_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:67: record 'channel.close' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:68: record 'channel.close_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:69: record 'channel.alert' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:70: record 'access.request' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:71: record 'access.request_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:72: record 'exchange.declare' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:73: record 'exchange.declare_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:74: record 'exchange.delete' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:75: record 'exchange.delete_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:76: record 'exchange.bind' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:77: record 'exchange.bind_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:78: record 'exchange.unbind' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:79: record 'exchange.unbind_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:80: record 'queue.declare' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:81: record 'queue.declare_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:82: record 'queue.bind' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:83: record 'queue.bind_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:84: record 'queue.purge' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:85: record 'queue.purge_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:86: record 'queue.delete' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:87: record 'queue.delete_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:88: record 'queue.unbind' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:89: record 'queue.unbind_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:90: record 'basic.qos' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:91: record 'basic.qos_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:92: record 'basic.consume' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:93: record 'basic.consume_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:94: record 'basic.cancel' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:95: record 'basic.cancel_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:96: record 'basic.publish' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:97: record 'basic.return' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:98: record 'basic.deliver' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:99: record 'basic.get' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:100: record 'basic.get_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:101: record 'basic.get_empty' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:102: record 'basic.ack' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:103: record 'basic.reject' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:104: record 'basic.recover_async' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:105: record 'basic.recover' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:106: record 'basic.recover_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:107: record 'basic.nack' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:108: record 'basic.credit' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:109: record 'basic.credit_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:110: record 'basic.credit_drained' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:111: record 'tx.select' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:112: record 'tx.select_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:113: record 'tx.commit' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:114: record 'tx.commit_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:115: record 'tx.rollback' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:116: record 'tx.rollback_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:117: record 'confirm.select' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:118: record 'confirm.select_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:119: record 'file.qos' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:120: record 'file.qos_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:121: record 'file.consume' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:122: record 'file.consume_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:123: record 'file.cancel' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:124: record 'file.cancel_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:125: record 'file.open' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:126: record 'file.open_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:127: record 'file.stage' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:128: record 'file.publish' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:129: record 'file.return' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:130: record 'file.deliver' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:131: record 'file.ack' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:132: record 'file.reject' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:133: record 'stream.qos' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:134: record 'stream.qos_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:135: record 'stream.consume' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:136: record 'stream.consume_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:137: record 'stream.cancel' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:138: record 'stream.cancel_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:139: record 'stream.publish' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:140: record 'stream.return' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:141: record 'stream.deliver' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:142: record 'dtx.select' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:143: record 'dtx.select_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:144: record 'dtx.start' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:145: record 'dtx.start_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:146: record 'tunnel.request' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:147: record 'test.integer' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:148: record 'test.integer_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:149: record 'test.string' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:150: record 'test.string_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:151: record 'test.table' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:152: record 'test.table_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:153: record 'test.content' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:154: record 'test.content_ok' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:156: record 'P_connection' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:157: record 'P_channel' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:158: record 'P_access' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:159: record 'P_exchange' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:160: record 'P_queue' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:161: record 'P_basic' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:162: record 'P_tx' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:163: record 'P_confirm' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:164: record 'P_file' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:165: record 'P_stream' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:166: record 'P_dtx' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:167: record 'P_tunnel' already defined
../../deps/rabbitmq_server-3.3.5/include/rabbit_framing.hrl:168: record 'P_test' already defined
make: *_\* [ebin/amimulator.app] Error 1

**Comment by busseyl** [comment]
Commented out line
amimulator_amqp.erl:19:%-include_lib("rabbitmq_server/include/rabbit_framing.hrl").

**Comment by danielfinke** [comment]
No longer relevant, amimulator_amqp has been removed from the code base
